### ISSUE TITLE
Fullrender after map purge only rendering chunks visited after purge 1.14.4

### ISSUE THREAD

**Issue by BlisteringSyko** [webbukkit/dynmap#2576]
Hello,

This past week, ive upgraded from 1.13.2 to 1.14.4, and last night i migrated to a newer system, so i increased the render quality to hi_boost_xhi as i now have the hdd space and cpu power for it. After changing the deftemplatesuffix and restarting the server, i purged each map with the commands. After i initiated a full render for flat, it completes very fast and upon inspection has only loaded a small area. It seems to load the spawn chunks for each world and each map. 1.14.4 chunks have also completely rendered ( we haven't explored much since i updated yet ) and when i visit 1.13.2 generated chunks, they only partially render.  

I'm not sure if this is the intended result of purging?

Ive also cleared my browser cache a few times, doesn't help.|
Also im hosting the web files in my Apache2 server.

Server is Paper build 156
Dynmap  
    core 3.0 beta 5 218
    plugin 3.0 beta 5 218

Thanks for all your hard work on this plugin for all this time.

**Comment by BlisteringSyko** [comment]
Here are some screen shots of the issue, first one is flat view, second is surface.  Newly generated chunks render properly, older chunks dont render till revisted, and they render in a checker pattern.

![08-06-2019  08-44-50-4204-AM](https://user-images.githubusercontent.com/32125693/62540870-90da3b80-b826-11e9-98b7-b05f2bc28cb6.jpeg)
![08-06-2019  08-45-00-6380-AM](https://user-images.githubusercontent.com/32125693/62540874-933c9580-b826-11e9-8f71-e000c798e938.jpeg)

**Comment by Tsoccerguy3** [comment]
You must be in TPS PAUSE , check with /dynmap pause none

Try lowering TPS from 18 to 8 , if you want rendering updates while players are online . I advise against it since it will be laggy . If you have 6 gigs and above wont matter . for 1.14.4 you should do as mush as you can before going live . 

add the switch --ForceUpgrade to force all maps and chunks to 1.14.4 . Then set your worldborders for each map and execute the commands /wb fill to pregenerate remaining chunks , then /wb trim to cleanup map.
When the maps are nice and tidy , execute /cleanlight world , for each world , this will fix all lighting
Then do /dynmap fullrender on each map

You are now ready for players . all this can be done on a local computer before you upload to a host.

Example of Sanacraft Dynmap on 1.14.4 http://81.19.208.15:25666/ 

plugins you will need
Dynmap
Worldborder
Lightcleaner

**Comment by BlisteringSyko** [comment]
Thanks, --forceupgrade seems to be doing the trick.

I purged all the maps again and now im doing a full render on all maps at the same time. Wish there was a way to make dynmap render faster, its only using about 40% cpu while rendering all 3 worlds.


I really dont think this was a Tps issue, even rendering everything at the same time, and flying around with an elytra like a crazy person generating new chunks, i cant even get it to drop below 20tps. 

Also, i dont use world border, i keep my worlds at max size so people can explore where ever they want. so pregernating everything is a no go, and i want my players to uncover the map them selves.. 

Thanks for the --forceupgrade tip though, i should have read the spigot post about upgrading from 1.13 to 1.14

Once everything is done rendering and i don't spot any bugs, ill come back and close this.

**Comment by BlisteringSyko** [comment]
Ok so the auto render, fullrender and updaterender aren't rendering chunks that were generated in 1.14.4 before i did --forceupgrade, at least not out around -984390, 64, -9588994,  however a radiusrender starting near that location with a 1,000,000 block radius seems to be doing the trick. 

It does seem to be working correctly on chunks generated after --forceupgrade and and on 1.14.4 chucks closer to 0 ,64, 0.


Im just gonna have to ask my users to let me know if they see any bad areas on the map so i can go and do a radius render

**Comment by generrosity** [comment]
Re-reading responses, and it looks like --forceupgrade + radiusrender did 99% of the trick. Also possible being hit by disjointed areas (as dynmap sometimes works like flood-fill I think)?

 I know The Dev has added some tricks into the more recent versions of Dynmap to help with upgrade edge cases. 

Always happy to reopen or add to the case, or feel free to paste the URL into a new ticket to reference it
